gitiquette.md

```diff
@@ -0,0 +1,153 @@
+committing to Spring:
+
+```bash
+$ git log --oneline -5 --author cbeams --before "Fri Mar 26 2009"
+```
+
+```
+e5f4b49 Re-adding ConfigurationPostProcessorTests after its brief removal in r814. @Ignore-ing the testCglibClassesAreLoadedJustInTimeForEnhancement() method as it turns out this was one of the culprits in the recent build breakage. The classloader hacking causes subtle downstream effects, breaking unrelated tests. The test method is still useful, but should only be run on a manual basis to ensure CGLIB is not prematurely classloaded, and should not be run as part of the automated build.
+2db0f12 fixed two build-breaking issues: + reverted ClassMetadataReadingVisitor to revision 794 + eliminated ConfigurationPostProcessorTests until further investigation determines why it causes downstream tests to fail (such as the seemingly unrelated ClassPathXmlApplicationContextTests) 147709f Tweaks to package-info.java files
+22b25e0 Consolidated Util and MutableAnnotationUtils classes into existing AsmUtils
+7f96f57 polishing Yikes. Compare that with these more recent commits from the same repository:
+```
+
+```bash
+$ git log --oneline -5 --author pwebb --before "Sat Aug 30 2014"
+```
+
+```
+5ba3db6 Fix failing CompositePropertySourceTests
+84564a0 Rework @PropertySource early parsing logic
+e142fd1 Add tests for ImportSelector meta-data
+887815f Update docbook dependency and generate epub
+ac8326d Polish mockito usage
+```
+
+Which would you rather read?
+
+The former varies wildly in length and form; the latter is concise and
+consistent. The former is what happens by default; the latter never happens by
+accident.
+
+While many repositories' logs look like the former, there are exceptions. The
+Linux kernel and git itself are great examples. Look at Spring Boot, or any
+repository managed by Tim Pope.
+
+The contributors to these repositories know that a well-crafted git commit
+message is the best way to communicate context about a change to fellow
+developers (and indeed to their future selves). A diff will tell you what
+changed, but only the commit message can properly tell you why. Peter Hutterer
+makes this point well:
+
+>Re-establishing the context of a piece of code is wasteful. We can't avoid it
+completely, so our efforts should go to reducing it [as much] as possible.
+Commit messages can do exactly that and as a result, a commit message shows
+whether a developer is a good collaborator. If you haven't given much thought to
+what makes a great git commit message, it may be the case that you haven't spent
+much time using git log and related tools. There is a vicious cycle here:
+because the commit history is unstructured and inconsistent, one doesn't spend
+much time using or taking care of it. And because it doesn't get used or taken
+care of, it remains unstructured and inconsistent.
+
+But a well-cared for log is a beautiful and useful thing. git blame, revert,
```

Comment:
Maybe put all git commands inside backticks, like `git blame`, to distinguish from text